File: app/src/main/java/ch/admin/bag/dp3t/checkin/startup/BootCompletedReceiver.java
package ch.admin.bag.dp3t.checkin.startup;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import ch.admin.bag.dp3t.storage.SecureStorage;
import ch.admin.bag.dp3t.checkin.utils.CrowdNotifierReminderHelper;
import ch.admin.bag.dp3t.checkin.utils.NotificationHelper;
import ch.admin.bag.dp3t.checkin.models.CheckInState;

public class BootCompletedReceiver extends BroadcastReceiver {

	@Override
	public void onReceive(Context context, Intent intent) {
		if (!Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
			return;
		}

		invalidateCheckIn(context);
	}

	private void invalidateCheckIn(Context context) {
		SecureStorage storage = SecureStorage.getInstance(context);
		CheckInState checkInState = storage.getCheckInState();
		if (checkInState == null) {
			return;
		}

		boolean checkedOut = CrowdNotifierReminderHelper.autoCheckoutIfNecessary(context, checkInState);

		if (!checkedOut) {
			long checkInTime = checkInState.getCheckInTime();
			NotificationHelper.getInstance(context).startOngoingNotification(checkInTime, checkInState.getVenueInfo());
			CrowdNotifierReminderHelper.set8HourReminder(checkInTime, context);
			CrowdNotifierReminderHelper.setAutoCheckOut(checkInTime, context);
			CrowdNotifierReminderHelper.setReminder(checkInTime + checkInState.getSelectedTimerOption().getDelayMillis(), context);
		}
	}

}

File: app/src/main/java/ch/admin/bag/dp3t/checkin/startup/BootCompletedReceiver.kt
package ch.admin.bag.dp3t.checkin.startup

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import ch.admin.bag.dp3t.checkin.utils.CrowdNotifierReminderHelper
import ch.admin.bag.dp3t.checkin.utils.NotificationHelper
import ch.admin.bag.dp3t.extensions.getAutoCheckoutDelay
import ch.admin.bag.dp3t.extensions.getCheckoutWarningDelay
import ch.admin.bag.dp3t.storage.SecureStorage

class BootCompletedReceiver : BroadcastReceiver() {
	override fun onReceive(context: Context, intent: Intent) {
		if (Intent.ACTION_BOOT_COMPLETED != intent.action) {
			return
		}
		invalidateCheckIn(context)
	}

	private fun invalidateCheckIn(context: Context) {
		val storage = SecureStorage.getInstance(context)
		val checkInState = storage.checkInState ?: return
		val checkedOut = CrowdNotifierReminderHelper.autoCheckoutIfNecessary(context, checkInState)
		if (!checkedOut) {
			val checkInTime = checkInState.checkInTime
			NotificationHelper.getInstance(context).startOngoingNotification(checkInTime, checkInState.venueInfo)
			CrowdNotifierReminderHelper.setCheckoutWarning(checkInTime, checkInState.venueInfo.getCheckoutWarningDelay(), context)
			CrowdNotifierReminderHelper.setAutoCheckOut(checkInTime, checkInState.venueInfo.getAutoCheckoutDelay(), context)
			CrowdNotifierReminderHelper.setReminder(checkInTime + checkInState.selectedReminderDelay, context)
		}
	}
}